generated/google_cloud_showcase_v1beta1/test/echo_test.dart

```diff
@@ -0,0 +1,155 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@TestOn('vm')
+library echo_test;
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:google_cloud_protobuf/protobuf.dart' show Any;
+import 'package:google_cloud_rpc/rpc.dart';
+import 'package:google_cloud_rpc/service_client.dart';
+import 'package:google_cloud_showcase_v1beta1/showcase.dart';
+import 'package:http/http.dart' as http;
+import 'package:path/path.dart' as p;
+import 'package:test_utils/insecure_proxy_http_client.dart';
+import 'package:test/test.dart';
+
+class AnyTypeName extends CustomMatcher {
+  AnyTypeName(dynamic matcher) : super('', 'typeName', matcher);
+
+  @override
+  Object? featureValueOf(dynamic actual) {
+    return (actual as Any).typeName;
+  }
+}
+
+class ShowcaseServer {
+  final Process _process;
+
+  static Future<void> _install() async {
+    // Install showcase rather than running it using `go run` because `go run`
+    // will then spawn showcase as a subprocess, which means that we don't be
+    // able to kill it.
+    final result = await Process.run('go', [
+      'install',
+      'github.com/googleapis/gapic-showcase/cmd/gapic-showcase@latest',
+    ]);
+    if (result.exitCode != 0) {
+      throw Exception('showcase installation failed ${result.stderr}');
+    }
+  }
+
+  static Future<String> _goBinaryPath() async {
+    final result = await Process.run('go', ['env', 'GOPATH']);
+    if (result.exitCode != 0) {
+      throw Exception('go env GOPATH failed ${result.stderr}');
+    }
+    return (result.stdout as String).trim();
+  }
+
+  static Future<String> _showcasePath() async {
+    return p.join(await _goBinaryPath(), 'bin', 'gapic-showcase');
+  }
+
+  ShowcaseServer._(this._process);
+
+  static Future<ShowcaseServer> start() async {
+    await _install();
+    final process = await Process.start(await _showcasePath(), ['run']);
+    final serverStarted = Completer<void>();
+    stderr.addStream(process.stderr);
+    process.stdin.close();
+    process.stdout
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((line) {
+          if (line.contains('Listening for REST connections')) {
+            serverStarted.complete();
+          }
+        });
+    await serverStarted.future;
```

Comment:
[nit] I don't think we need to mix `async` with a manual `Completer` here.

```suggestion
    stderr.addStream(process.stderr);
    process.stdin.close();
    await process.stdout
        .transform(utf8.decoder)
        .transform(const LineSplitter())
        .firstWhere((line) => line.contains('Listening for REST connections'));
```

Reply:
That won't work because it will end up cancelling the `stdout` stream subscription, which will close `stdout` for the subprocess, which will cause writes to `stdout` to fail, which will cause the subprocess to exit.